https://gitlab.gnome.org/GNOME/gnome-online-accounts/-/commit/f10c48ee114f719c63a8eabccdfb401a14219f46
https://bugs.gentoo.org/882625

From: Emmanuele Bassi <ebassi@gnome.org>
Date: Wed, 5 Oct 2022 22:15:37 +0100
Subject: [PATCH] build: Use the appropriate dependency object

Just using `link_with` will not ensure that the GOA web extension shared
module is properly built against libgoa-backend and libgoa; the
generated headers must be transitively available by the time we build
the shared module.

Fixes: #226
--- a/src/goabackend/meson.build
+++ b/src/goabackend/meson.build
@@ -157,9 +157,8 @@ libgoa_web_extension = shared_module(
   'goawebextension',
   libgoawebextension_sources,
   include_directories: common_incs + [goa_inc],
-  dependencies: deps,
+  dependencies: [deps, libgoa_backend_dep],
   c_args: cflags,
-  link_with: libgoa_backend,
   install: true,
   install_dir: join_paths(goa_pkglibdir, 'web-extensions')
 )
GitLab
